docs/Tutorial_DAMP.ipynb

```diff
@@ -0,0 +1,569 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #67.            PL[start : stop - m + 1] = np.minimum(PL[start : stop - m + 1], D)
</pre><p>Previously, we had the variable <code>pruned</code> , which was a boolean vector, where <code>pruned[i]</code> is pruned if the i-th subsequence is pruned. And, instead of line above (i.e. line #67), we had:</p><p> </p><pre class="ql-syntax" spellcheck="false">mask = np.argwhere(D &lt; bsf) + start
pruned[mask] = True
</pre><p> </p><p>Recall that the paper's original algorithm does not compute <code>PL[i]</code> if <code>pruned[i]</code> is True. It just skips it. In such case, the original algorithm set <code>PL[i]</code> as follows:   </p><p> </p><pre class="ql-syntax" spellcheck="false">PL[i] = PL[i-1]
</pre><p> </p><p>which makes it difficult to find the correct index of discord. The MATALB code does a hack instead as follows:</p><pre class="ql-syntax" spellcheck="false">PL[i] = PL[i-1] - 0.000001 
</pre><p> </p><p>and this does not seem to be a clean solution.</p><p> </p><p>So, instead, I am updating the (approximate) matrix profile <code>PL</code> by using the computed distance profile D . This helps us to avoid the hack, and I think It should not be computationally more expensive compared to <code>np.argwhere(D &lt; bfs)</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/TDAmeritrade/stumpy/pull/920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>